include/onnx-mlir/Compiler/OMCompilerTypes.h

```diff
@@ -5,42 +5,44 @@
 #ifndef ONNX_MLIR_OMCOMPILERTYPES_H
 #define ONNX_MLIR_OMCOMPILERTYPES_H
 
+#ifdef __cplusplus
 namespace onnx_mlir {
+#endif
 
 /* Type of compiler emission targets */
-enum EmissionTargetType {
+typedef enum {
```

Comment:
Is that the preferred way? In the OnnxMlirRuntime, they use this:
```
#ifndef __cplusplus
typedef struct OMTensor OMTensor;
#endif
```
namely they use the typedef only for C. I am fine with either versions, want a clarification on which one is better, if there is any differences between the two.

Reply:
Yes. In C the typedef declaration allows the user to write `EmissionTargetType tgtType;` while the way it was before would force C users to write `enum EmissionTargetType tgtType;`. For C++ users there is not difference between the 2 declarations.